jetty-io/src/main/java/org/eclipse/jetty/io/ArrayRetainableByteBufferPool.java

```diff
@@ -191,9 +191,19 @@ public RetainableByteBuffer acquire(int size, boolean direct)
         return buffer;
     }
 
+    protected ByteBuffer allocate(int capacity)
+    {
+        return ByteBuffer.allocate(capacity);
+    }
+
+    protected ByteBuffer allocateDirect(int capacity)
+    {
+        return ByteBuffer.allocateDirect(capacity);
+    }
```

Comment:
Note that retainable pool takes buffers from the normal pool, so that can reduce wastage as the retainable pool grows.   I have yet to implement release, but will do so.